### PR TITLE
CLOUDP-167795: Migrate Custom Database Roles CLI stores to v2 sdk

### PR DESCRIPTION
**Pull request by swattyT**
## Proposed changes

_Jira ticket:_ CLOUDP-# [CLOUDP-167795](https://jira.mongodb.org/browse/CLOUDP-167795)
EVG Patch: https://spruce.mongodb.com/version/6422f5dae3c3319349d1fd5b/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by swattyT — COMMENTED** [review]
Local testing:

`➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles create role-swati --privilege UPDATE@db.collection --inheritedRole enableSharding@admin --projectId 641deaf951ed5c679239b1a1 -o=json
{
  "actions": [
    {
      "action": "UPDATE",
      "resources": [
        {
          "cluster": false,
          "collection": "collection",
          "db": "db"
        }
      ]
    }
  ],
  "inheritedRoles": [
    {
      "db": "admin",
      "role": "enableSharding"
    }
  ],
  "roleName": "role-swati"
}

➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles ls -o=json
Error: required flag(s) "projectId" not set

➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles ls --projectId 641deaf951ed5c679239b1a1 -o=json
[
  {
    "actions": [
      {
        "action": "FLUSH_ROUTER_CONFIG",
        "resources": [
          {
            "cluster": true,
            "collection": "",
            "db": ""
          }
        ]
      },
      {
        "action": "",
        "resources": [
          {
            "cluster": true,
            "collection": "",
            "db": ""
          }
        ]
      }
    ],
    "inheritedRoles": [
      {
        "db": "admin",
        "role": "enableSharding"
      }
    ],
    "roleName": "role-sw"
  },
  {
    "actions": [
      {
        "action": "UPDATE",
        "resources": [
          {
            "cluster": false,
            "collection": "collection",
            "db": "db"
          }
        ]
      }
    ],
    "inheritedRoles": [
      {
        "db": "admin",
        "role": "enableSharding"
      }
    ],
    "roleName": "role-swati"
  }
]

➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles ls --projectId 641deaf951ed5c679239b1a1        
NAME         ACTION                INHERITED ROLES   DB      COLLECTION   CLUSTER 
role-sw      FLUSH_ROUTER_CONFIG   N/A               N/A     N/A          true
role-sw                            N/A               N/A     N/A          true
role-sw      N/A                   enableSharding    admin   N/A          N/A 
role-swati   UPDATE                N/A               db      collection   N/A   
role-swati   N/A                   enableSharding    admin   N/A          N/A

➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles describe role-swati  --projectId 641deaf951ed5c679239b1a1 -o=json
{
  "actions": [
    {
      "action": "UPDATE",
      "resources": [
        {
          "cluster": false,
          "collection": "collection",
          "db": "db"
        }
      ]
    }
  ],
  "inheritedRoles": [
    {
      "db": "admin",
      "role": "enableSharding"
    }
  ],
  "roleName": "role-swati"
}

➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles describe role-swati  --projectId 641deaf951ed5c679239b1a1        
NAME         ACTION   DB    COLLECTION   CLUSTER 
role-swati   UPDATE   db    collection   N/A   


➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles update role-swati --privilege LIST_SESSIONS --privilege UPDATE@db2.collection --append --projectId 641deaf951ed5c679239b1a1 -o=json
{
  "actions": [
    {
      "action": "LIST_SESSIONS",
      "resources": [
        {
          "cluster": true,
          "collection": "",
          "db": ""
        }
      ]
    },
    {
      "action": "UPDATE",
      "resources": [
        {
          "cluster": false,
          "collection": "collection",
          "db": "db2"
        },
        {
          "cluster": false,
          "collection": "collection",
          "db": "db"
        }
      ]
    }
  ],
  "inheritedRoles": [
    {
      "db": "admin",
      "role": "enableSharding"
    }
  ],
  "roleName": "role-swati"
}

➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles update role-swati --privilege LIST_SESSIONS --projectId 641deaf951ed5c679239b1a1 -o=json
{
  "actions": [
    {
      "action": "LIST_SESSIONS",
      "resources": [
        {
          "cluster": true,
          "collection": "",
          "db": ""
        }
      ]
    }
  ],
  "roleName": "role-swati"
}

➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles update role-swati --privilege LIST_SESSIONS --projectId 641deaf951ed5c679239b1a1        
Custom database role 'role-swati' successfully updated.

➜  mongodb-atlas-cli git:(CLOUDP-167795) ✗ bin/atlas customDbRoles delete  role-swati --projectId 641deaf951ed5c679239b1a1        
? Are you sure you want to delete: role-swati Yes
Custom database role 'role-swati' deleted
`

File: go.sum [hunk 1]
```diff
@@ -401,6 +401,8 @@ github.com/yuin/goldmark v1.3.5/go.mod h1:mwnBkeHKe2W/ZEtQ+71ViKU8L12m81fl3OWwC1
 github.com/yuin/goldmark v1.4.13/go.mod h1:6yULJ656Px+3vBD8DxQVa3kxgyrAnzto9xy5taEt/CY=
 go.mongodb.org/atlas v0.23.2-0.20230323121922-76f1516f3ff9 h1:5/GxsRQFvAW9UZD/GAcjQMQAfEWTjvdv1INYjqoL53w=
 go.mongodb.org/atlas v0.23.2-0.20230323121922-76f1516f3ff9/go.mod h1:r5HHjwxout7CJwjkAcJt+c2yA43rrB6LXidPk3pyeuo=
+go.mongodb.org/atlas v0.23.2-0.20230328111225-e40463c98c4e h1:Uy+DI9phtg3/M28DYH6u8Qvhh033yUhYlWlUqI2x2Iw=
```
Comment (gssbzn) [inline go.sum:404]: can you run `go mod tidy` when updating the client?

File: internal/store/database_roles.go [hunk 2]
```diff
@@ -59,40 +60,44 @@ func (s *Store) CreateDatabaseRole(groupID string, role *atlas.CustomDBRole) (*a
...
+		_, err := s.clientv2.CustomDatabaseRolesApi.DeleteCustomDatabaseRole(s.ctx, groupID, roleName).Execute()
 		return err
 	default:
 		return fmt.Errorf("%w: %s", errUnsupportedService, s.service)
 	}
 }
 
 // DatabaseRoles encapsulate the logic to manage different cloud providers.
-func (s *Store) DatabaseRoles(projectID string, opts *atlas.ListOptions) (*[]atlas.CustomDBRole, error) {
+func (s *Store) DatabaseRoles(projectID string, _ *atlas.ListOptions) ([]atlasv2.CustomDBRole, error) {
```
Comment (gssbzn) [inline internal/store/database_roles.go:71]: if not needed let's delete it
```suggestion
func (s *Store) DatabaseRoles(projectID string) ([]atlasv2.CustomDBRole, error) {
```

File: go.sum (hunk 1 above)
Comment (swattyT) [reply go.sum:402]: Done. Thank You!

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**